pkgs/development/python-modules/torch/default.nix

```diff
@@ -611,6 +611,23 @@ buildPythonPackage rec {
       install_name_tool -change @rpath/libc10.dylib $lib/lib/libc10.dylib $lib/lib/libshm.dylib
     '';
 
+  # See https://github.com/NixOS/nixpkgs/issues/296179
+  #
+  # This is a quick hack to add `libnvrtc` to the runpath so that torch can find
+  # it when it is needed at runtime.
+  extraRunpaths = [ "${lib.getLib cudaPackages.cuda_nvrtc}/lib" ];
+  postPhases = lib.optionals (cudaSupport && stdenv.isLinux) [ "postPatchelfPhase" ];
```

Comment:
Nit: I don't think `cudaSupport` belongs here. THe reasoning for `isLinux` was that it's lazy (and not entirely correct) way of saying "`postPatchelfPhase` only makes sense on systems that use ELF". Not that it matters a whole lot, because I at least still hope to see  a more generic hook introduced, which would allow removing ad hoc code from torch and other derivations

Reply:
So without `cudaSupport` here, my understanding is that for non-cuda build on Linux the hook is still going to perform the patch. Do you mean this is ok?

Reply:
Yes, it'll just be a no-op when `extraRunpaths` is empty